wireguard/metrics.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wireguard
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/projectcalico/felix/netlinkshim"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+const (
+	labelHostname      = "hostname"
+	labelPublicKey     = "public_key"
+	labelInterfaceName = "iface"
+	labelListenPort    = "listen_port"
+	labelPeerKey       = "peer_key"
+	labelPeerEndpoint  = "peer_endpoint"
+)
+
+var _ prometheus.Collector = (*Metrics)(nil)
+
+const (
+	wireguardMetaFQName   = "wireguard_meta"
+	wireguardMetaHelpText = "wireguard interface and runtime metadata"
+
+	wireguardLatestHandshakeIntervalFQName   = "wireguard_latest_handshake_seconds"
+	wireguardLatestHandshakeIntervalHelpText = "wireguard interface latest handshake unix timestamp in seconds to a peer"
+
+	wireguardBytesSentFQName   = "wireguard_bytes_sent"
+	wireguardBytesSentHelpText = "wireguard interface total outgoing bytes from peer"
+
+	wireguardBytesRcvdFQName   = "wireguard_bytes_rcvd"
+	wireguardBytesRcvdHelpText = "total incoming bytes from peer"
```

Comment:
`wireguardBytesSentHelpText` and `wireguardBytesRcvdHelpText` are slightly inconsistent with each other, one begins with "wireguard interface " and the other doesn't.

Reply:
good catch! also the logic of one of those help texts are wrong.. thanks!